### PR TITLE
fix(es/parser): emit error if await used as ident in class static ini…

### PR DESCRIPTION
**Pull request by iantanwx**
…tializer

**Description:**

This PR fixes https://github.com/swc-project/swc/issues/8374

**BREAKING CHANGE:**

**Related issue (if exists):**

https://github.com/swc-project/swc/issues/8374

### CODE REVIEW

**Comment by magic-akari** [comment]
This case should be added.

```JavaScript
class Foo {
    static {
        function foo() {
            var await = 'bar';
        }
    }
}
```

**Comment by iantanwx** [comment]
> This case should be added.
> 
> ```js
> class Foo {
>     static {
>         function foo() {
>             var await = 'bar';
>         }
>     }
> }
> ```

Thanks for this, I've added it.

**Review by kdy1 — DISMISSED** [review]
Thanks!

---

swc-bump:
 - swc_ecma_parser

File: crates/swc_ecma_parser/src/parser/stmt.rs [hunk 1]
```diff
@@ -2407,6 +2407,19 @@ export default function waitUntil(callback, options = {}) {
         test_parser(src, Syntax::Es(Default::default()), |p| p.parse_expr());
     }
 
+    #[test]
+    #[should_panic(expected = "Expected ident")]
```
Comment (magic-akari) [inline crates/swc_ecma_parser/src/parser/stmt.rs:2411]: No. It should not panic. It's valid JavaScript.

Comment (iantanwx) [reply crates/swc_ecma_parser/src/parser/stmt.rs]: @magic-akari i didn't realised 😓  

fixed it, but i'm not sure if the solution accounts for all corner cases...could you take a look?

Comment (magic-akari) [reply crates/swc_ecma_parser/src/parser/stmt.rs]:
```JavaScript
class Foo {
    static {
        var await = "bar"; // SyntaxError: Unexpected reserved word

        function foo() {
            var await = "bar"; // ok

            class Bar {
                static {
                    var await = "bar"; // SyntaxError: Unexpected reserved word
                }
            }
        }
    }
}
```

I recommend setting the variable `in_static_block` to `false` upon entering the function body. 
Please note that I cannot guarantee that all scenarios have been considered.

Comment (iantanwx) [reply crates/swc_ecma_parser/src/parser/stmt.rs:2411]: Fixed, your suggested solution seems much better to me.

Comment (iantanwx) [reply crates/swc_ecma_parser/src/parser/stmt.rs:2411]: @magic-akari bump

**Review by magic-akari — APPROVED** [review]

**Review by sweetliquid — APPROVED** [review]

**Pull request merged**